### ISSUE TITLE
ValidationSummary doesn't display property errors if ModelState and ModelMetadata meets the specific conditions

### ISSUE THREAD

**Issue by lukaskabrt** [aspnet/AspNetCore#4831]
### Title
ValidationSummary doesn't display property errors if ModelState and ModelMetadata meets the specific conditions

### Functional impact
If user submits a form, he doesn't see validation messages in the validation summary

### Minimal repro steps

1. Create a model, that implements `IEnumerable` and has `ID` property
```
    public class FormMetadata : IEnumerable<FieldMetadata> {
        private List<FieldMetadata> _fields = new List<FieldMetadata>();
        public int ID { get; set; }

        public IEnumerator<FieldMetadata> GetEnumerator() {
            return _fields.GetEnumerator();
        }

        IEnumerator IEnumerable.GetEnumerator() {
            return _fields.GetEnumerator();
        }
    }

    public class FieldMetadata {
    }
```
2. Create an action in a controller, that has `id` parameter
3. In the action add validation error to the property that isn't part of the model
```
[HttpPost, ActionName("Index")]
public IActionResult IndexPOST(int id) {
    var model = new FormMetadata();
    ModelState.AddModelError("MyForm.MyField", "Something went wrong.");

    return View(model);
}
```

4. Run project and send the form

See example project. To reproduce the error just submit the form
[validation-summary-issue.zip](https://github.com/aspnet/Mvc/files/924488/validation-summary-issue.zip)

### Expected result
I expect all validation errors to appear in the validation summary.

### Actual result
The empty validation summary is generated.

```
<div class="validation-summary-errors" data-valmsg-summary="true">
    <ul>
        <li style="display:none"></li>
    </ul>
</div>
```

### Further technical details
My use case for this scenario: Model contains metadata for a dynamically created form. In controller I add validation errors for dynamically created fields.

I tracked the issue down to the `GetModelStateList` method in the `ValidationHelpers` class. `ModelStateEntry` is created for `FormMetadata.ID`, another one is created for `id` parameter from the controller and because of `if (entries.Count < modelStateDictionary.Count)` check the error message for `MyForm.MyField` isn't created.

**Comment by kishanAnem** [comment]
Hi @dougbu 

https://github.com/aspnet/AspNetCore/blob/321327f84b2b22dcff2e9beb06a9a64236c5cced/src/Mvc/src/Microsoft.AspNetCore.Mvc.ViewFeatures/ValidationHelpers.cs#L101

I've changed above line to like this. I think this may be fix. Please let me know isn't it fix?

```
   if (propertyModelStateEntry != null && !orderedModelStateEntries.Contains(propertyModelStateEntry))
      {
            Visit(propertyModelStateEntry, propertyMetadata, orderedModelStateEntries);
       }
```

**Comment by dougbu** [comment]
@kishanAnem we plan to release a fix for this issue in preview 3. When I get to it, I'll look at your suggestion. But, if you prefer to get a fix in sooner (at least to MVC nightlies), feel free to open a PR.

(Please reference this issue in your PR if you choose to open one. A comment here may help bring it to my attention.)

**Comment by kishanAnem** [comment]
Hi @dougbu 

I've opened a PR.

**Comment by dougbu** [comment]
Fixed in c1ce7b9a77 with thanks to @kishanAnem

**Comment by kishanAnem** [comment]
I should say thanks @dougbu for your help and mentoring.
